x-py.md

```diff
@@ -0,0 +1,392 @@
+---
+layout: default
+title: Building rustc with x.py; The Rust Forge
+---
+
+# What is `x.py`?
+
+`x.py` is the script used to orchestrate the tooling in the `rustc` repository.
+It is the script that can build docs, run tests, and compile `rustc`. It is the
+now preferred way to build `rustc` and it replaces the old makefiles from
+before. Below are the different ways to utilize `x.py` in order to effectively
+deal with the repo for various common tasks.
+
+# Table of Contents
+- [Building `rustc`](#building-rustc)
+  - [Build everything](#build-everything)
+  - [Build different stages](#build-different-stages)
+  - [Build specific components](#build-specific-components)
+  - [Other Flags](#other-flags)
+- [Testing rustc](#testing-rustc)
+  - [Run all tests](#run-all-tests)
+  - [Run specific tests](#run-specific-tests)
+  - [Other Flags](#other-flags-2)
+- [Benchmarking rustc](#benchmarking-rust)
+  - [Other Flags](#other-flags-3)
+- [Documenting rustc](#documenting-rustc)
+  - [Document everything](#document-everything)
+  - [Document specific components](#document-specific-components)
+  - [Document internal rustc items](#document-internal-rustc-items)
+  - [Other Flags](#other-flags-4)
+- [Cleaning out build directories]()
+  - [Other Flags](#other-flags-5)
+- [Build distribution artifacts]()
+  - [Other Flags](#other-flags-6)
+- [Install distribution artifacts]()
+  - [Other Flags](#other-flags-7)
+
+# Building `rustc`
+If you want to build the compiler to add new features you'll need to run one of
+the build commands. Note that the first time you build the compiler it will take
+a longer time to do so since you'll need to compile the entire LLVM library.
+However, subsequent builds won't be as long unless LLVM is updated.
+
+### Build everything
+
+```bash
+./x.py build
+```
+
+This command will build the compiler. Because Rust is a bootstrapping compiler
+(the Rust compiler is written in Rust) this will involve compiling the compiler
+twice. The first compilation creates the "stage 1" compiler and the second
+compilation creates the "stage 2" compiler. The "stage 2" compiler is typically
+considered the final compiler and is what tests are run with and what you will
+likely interact with. In some situations, though, the "stage 1" compiler may be
```

Comment:
I'm not sure if we should say that the stage2 compiler is what you'll be interacting with most likely -- I'd guess most work done by developers of rustc happens at the stage1 (possible with std) level. It also feels like we're inconsistent since we later say that stage 1 is "typically enough to test out all of your changes to the compiler"